ofrak_core/test_ofrak/components/test_gzip_component.py

```diff
@@ -1,87 +1,103 @@
-import os
-import subprocess
-import tempfile
-from gzip import GzipFile
-from io import BytesIO
-
+import zlib
+import gzip
+from pathlib import Path
+from asyncio import create_subprocess_exec
+from asyncio.subprocess import PIPE
+from typing import Tuple
+from unittest.mock import patch
+
+from ofrak.component.abstract import ComponentSubprocessError
 import pytest
 
-from ofrak import OFRAKContext
+from ofrak.ofrak_context import OFRAKContext
 from ofrak.resource import Resource
 from ofrak.core.gzip import GzipData
 from pytest_ofrak.patterns.compressed_filesystem_unpack_modify_pack import (
     CompressedFileUnpackModifyPackPattern,
 )
-from pytest_ofrak.patterns.unpack_modify_pack import UnpackModifyPackPattern
+
+ASSETS_DIR = Path(__file__).parent / "assets"
 
 
-class TestGzipUnpackModifyPack(CompressedFileUnpackModifyPackPattern):
+@pytest.fixture(
+    autouse=True,
+    scope="module",
+    params=[
+        (ASSETS_DIR / "hello_world", ASSETS_DIR / "hello_ofrak", False),
+        (ASSETS_DIR / "random8M", ASSETS_DIR / "random8M_modified", True),
+    ],
+    ids=["hello world", "<random 8MB data>"],
+)
+def gzip_test_input(request):
+    initial_path, repacked_path, expect_pigz = request.param
+    with open(initial_path, "rb") as initial_file:
+        initial_data = initial_file.read()
+    with open(repacked_path, "rb") as repacked_file:
+        expected_repacked_data = repacked_file.read()
+    return (initial_data, expected_repacked_data, expect_pigz)
+
+
+class GzipUnpackModifyPackPattern(CompressedFileUnpackModifyPackPattern):
+    """
+    Template for tests that test different inputs the gzip component should support
+    unpacking.
+    """
+
+    EXPECT_PIGZ: bool
     expected_tag = GzipData
 
-    @pytest.fixture(autouse=True)
-    def create_test_file(self, tmpdir):
-        d = tmpdir.mkdir("gzip")
-        fh = d.join("hello.gz")
-        result = BytesIO()
-        with GzipFile(fileobj=result, mode="w") as gzip_file:
-            gzip_file.write(self.INITIAL_DATA)
-        fh.write_binary(result.getvalue())
+    def write_gzip(self, gzip_path: Path):
+        raise NotImplementedError
```

Comment:
This should be `raise NotImplementedError()`: https://ofrak.com/docs/contributor-guide/getting-started.html#raising-exceptions.

Reply:
Should it also be marked as an `@abstractmethod`?